src/server/socketHandlers/scConnectionHandler.js

```diff
@@ -2,23 +2,49 @@ import scSubscribeHandler from './scSubscribeHandler';
 import scUnsubscribeHandler from './scUnsubscribeHandler';
 import scGraphQLHandler from './scGraphQLHandler';
 import {REFRESH_JWT_AFTER} from 'server/utils/serverConstants';
+import getRetink from 'server/database/rethinkDriver';
+import isObject from 'universal/utils/isObject';
+import jwtDecode from 'jwt-decode';
 
+// we do this otherwise we'd have to blacklist every token that ever got replaced & query that table for each query
+const istmsValid = (tmsFromDB, tmsFromToken) => {
+  if (tmsFromDB.length !== tmsFromToken.length) return false;
+  for (let i = 0; i < tmsFromDB.length; i++) {
+    if (tmsFromDB[i] !== tmsFromToken[i]) return false;
+  }
+  return true;
+};
 export default function scConnectionHandler(exchange) {
   return async function connectionHandler(socket) {
+    const r = getRetink();
     const authToken = socket.getAuthToken();
     const now = new Date();
     const tokenExpiration = new Date(authToken.exp * 1000);
     const timeLeftOnToken = tokenExpiration - now;
-    if (timeLeftOnToken < REFRESH_JWT_AFTER) {
+    // if the user was booted from the team, give them a new token
+    const tms = await r.table('User').get(authToken.sub)('tms');
+    const tmsIsValid = istmsValid(tms, authToken.tms);
```

Comment:
when we connect, we now check the validity of the token. it's a little extra cost, but doing it once per connection is a fair price.